core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java

```diff
@@ -250,14 +252,196 @@ In practice, the stop train will require the route in front of him (zone and swi
             var simResultAWithStop =
                     simpleSim(fullInfra, pathPropsA, chunkPathA, 0, Double.POSITIVE_INFINITY, List.of(stop));
 
-            var conflicts = ConflictsKt.detectConflicts(List.of(
-                    convertRequirements(0L, 0.0, simResultAWithStop.train),
-                    convertRequirements(1L, 0.0, simResultB.train)));
+            var reqA = convertRequirements(0L, 0.0, simResultAWithStop.train);
+            var reqB = convertRequirements(1L, 0.0, simResultB.train);
+            var conflicts = ConflictsKt.detectConflicts(List.of(reqA, reqB));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == ROUTING));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == SPACING));
         }
     }
 
+    /*
+    Context: both trains start in station and the requirements on the block protecting the switch at station exit may
+    conflict.
+    This block is protected by 2 signals (first will be yellow, last and closest will be red).
+    Test for trains starting after the first signal protecting block with the switch.
+    Also, the stopping train stops either (all combinations):
+    * on stop-signal or on open signal
+    * while seeing the last signal protecting block or before seeing it
+
+    Stopping on open signal and before seeing any signal protecting the block is quite an undefined behavior (train
+    is starting after first signal).
+    In current implementation, the considered signal is not really the limiting one, but the first signal seen by the
+    train that is protecting the block.
+    So one might consider that train should require resource at its very beginning, but we actually go for waiting
+    to see any signal.
+     */
+    @ParameterizedTest
+    @CsvSource({
+        "false, 100, true, true",
+        "true, 100, false, false",
+        "false, -100, false, false",
+        "true, -100, false, false",
+    })
+    public void conflictHandlingWithTrainStartingDuringRequirementTrigger(
+            boolean onStopSignal,
+            double distanceAfterLastSignalSight,
```

Comment:
It's unclear what this parameter does from it's name. I'd rather have a boolean `seeingEntrySignal` or something like that.

Reply:
Switch done in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/893a440284a753f05b51ecca1b87991625dd6ec8 (same for limitingSignal in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/db81e2e0fcc727ddc13b248276126ce15731b326).